src/agent/src/rpc.rs

```diff
@@ -1634,29 +1635,37 @@ fn append_guest_hooks(s: &Sandbox, oci: &mut Spec) -> Result<()> {
     Ok(())
 }
 
-// Check is the container process installed the
+// Check if the container process installed the
 // handler for specific signal.
-fn is_signal_handled(pid: pid_t, signum: u32) -> bool {
-    let sig_mask: u64 = 1u64 << (signum - 1);
-    let file_name = format!("/proc/{}/status", pid);
+fn is_signal_handled(proc_status_file: &str, signum: u32) -> bool {
+    if signum > 64 {
```

Comment:
Can you add a comment here explaining what this test is doing? Nominally, it appears to be just checking against some arbitrary maximum value for a signal number. But that's not the case - this line is "bound" with the bit shift done on line 1660.

Alternatively, you could do something like this (not: no `\n` required on the log calll):

```rust
// Ensure invalid signal number doesn't break the shift logic
let shift_count: u64 = if signum > 64 {
    warn!(sl!(), "received invalid signum {}", signum);
    return false;
} else {
    signum - 1
};

// ...

let sig_mask: u64 = 1u64 << shift_count;
```

Reply:
Thanks @jodh-intel, good suggestion. I left the `signum - 1` where it was, because moving it above the process file check and `signum == 0` check led to trying to put a negative number in an unsigned integer, as we are allowing `0` as a valid signum.

I guess I could have moved the `signum > 64` check below the process file check and `signum == 0` check, but I think having the validation at the top of function makes the most sense.

Reply:
+1 for handling the checks at the start of the function. The problem is having multiple checks on the same variable spread out through that function. You could handle all those checks first with something like this:

```diff
diff --git a/src/agent/src/rpc.rs b/src/agent/src/rpc.rs
index 488e5edf2..dc85db043 100644
--- a/src/agent/src/rpc.rs
+++ b/src/agent/src/rpc.rs
@@ -1639,11 +1639,21 @@ fn append_guest_hooks(s: &Sandbox, oci: &mut Spec) -> Result<()> {
 // Check if the container process installed the
 // handler for specific signal.
 fn is_signal_handled(proc_status_file: &str, signum: u32) -> bool {
-    // Ensure invalid signum won't break bit shift logic
-    if signum > 64 {
+    let shift_count: u64 = if signum == 0 {
+        // signum 0 is used to check for process liveness.
+        // Since that signal is not part of the mask in the file, we only need
+        // to know if the file (and therefore) process exists to handle
+        // that signal.
+        return fs::metadata(proc_status_file).is_ok();
+    } else if signum > 64 {
+        // Ensure invalid signal number doesn't break the shift logic.
         warn!(sl!(), "received invalid signum {}", signum);
         return false;
-    }
+    } else {
+        (signum - 1).into()
+    };
+
+    let sig_mask: u64 = 1u64 << shift_count;

     // Open the file in read-only mode (ignoring errors).
     let file = match File::open(proc_status_file) {
@@ -1654,12 +1664,6 @@ fn is_signal_handled(proc_status_file: &str, signum: u32) -> bool {
         }
     };

-    // signum 0 is used to check for process liveness
-    if signum == 0 {
-        return true;
-    }
-
-    let sig_mask: u64 = 1u64 << (signum - 1);
     let reader = BufReader::new(file);

     // Read the file line by line using the lines() iterator from std::io::BufRead.
```

Reply:
@jodh-intel Thanks, that all makes sense. I'll keep it in mind for the future.